routes/auth/oauth2callback.ts

```diff
@@ -1,33 +1,19 @@
 import { Handlers } from "$fresh/server.ts";
-import { deleteCookie, getCookies, setCookie } from "$std/http/cookie.ts";
-import { getAndDeleteOauthSession, setUserWithSession } from "🛠️/db.ts";
+import { getSessionTokens, handleCallback } from "kv_oauth";
+import { client } from "🛠️/kv_oauth.ts";
 import { getAuthenticatedUser } from "🛠️/github.ts";
-import { oauth2Client } from "🛠️/oauth.ts";
-import { User } from "🛠️/types.ts";
+import { setUserWithSession } from "🛠️/db.ts";
+import type { User } from "🛠️/types.ts";
+import { getSetCookies } from "$std/http/cookie.ts";
 
 export const handler: Handlers = {
   async GET(req) {
-    const cookies = getCookies(req.headers);
-    const oauthSessionCookie = cookies["oauth-session"];
-    if (!oauthSessionCookie) {
-      return new Response("Missing oauth session", {
-        status: 400,
-      });
-    }
-    const oauthSession = await getAndDeleteOauthSession(oauthSessionCookie);
-    if (!oauthSession) {
-      return new Response("Missing oauth session", {
-        status: 400,
-      });
-    }
-    const { state, codeVerifier } = oauthSession;
-    const tokens = await oauth2Client.code.getToken(req.url, {
-      state,
-      codeVerifier,
-    });
-    const ghUser = await getAuthenticatedUser(tokens.accessToken);
+    const resp = await handleCallback(req, client);
+    const [{ value: session }] = getSetCookies(resp.headers);
+
+    const tokens = await getSessionTokens(session);
+    const ghUser = await getAuthenticatedUser(tokens!.accessToken);
```

Comment:
This seems hacky... Perhaps, `handleCallback()` should return `response`, `sessionId` and `tokens`.

Reply:
Done.